lib/api/index.js

```diff
@@ -0,0 +1,202 @@
+      console.log(`login=${data.viewer.login}, isEmployee=${data.viewer.isEmployee}, viewerPermission=${data.viewer.organization.repository.viewerPermission}`)
+      res.status(401)
+      return res.json({
+        error: 'Unauthorized',
+        message: 'Token provided does not have `WRITE` access to the @github/ecosystem-primitives repo.'
+      })
+    }
+
+    console.log(`Staff routes accessed:`)
+    console.log(`login=${data.viewer.login}, isEmployee=${data.viewer.isEmployee}, viewerPermission=${data.viewer.organization.repository.viewerPermission}`)
+
+    next()
+  } catch (err) {
+    console.log({err})
+    return res.sendStatus(500)
+  }
+})
+
+app.get('/', (req, res) => {
+  res.send({})
+})
+
+app.get('/:installationId', async (req, res) => {
+  const { Subscription } = require('../models')
+  const { installationId } = req.params
+  const { client } = res.locals
+  try {
+    const subscriptions = await Subscription.getAllForInstallation(installationId)
+    if (!subscriptions.length) {
+      return res.sendStatus(404)
+    }
+    const { jiraHost } = subscriptions[0].dataValues
+    const installations = await Promise.all(subscriptions.map(subscription => getInstallation(client, subscription)))
+    const connections = installations
+      .filter(response => !response.error)
+      .map(data => ({
+        ...data,
+        isGlobalInstall: data.repository_selection === 'all',
+        updated_at: format(data.updated_at, 'MMMM D, YYYY h:mm a'),
+        syncState: data.syncState
+      }))
+
+    const failedConnections = installations.filter(response => {
+      console.log(response.error)
+      return response.error
+    })
+    res.json({
+      host: jiraHost,
+      installationId,
+      connections,
+      failedConnections,
+      hasConnections: connections.length > 0 || failedConnections.length > 0,
+      repoSyncState: `${req.protocol}://${req.get('host')}/api/${installationId}/repoSyncState.json`
+    })
+  } catch (err) {
+    console.log(err)
+    res.status(500)
+    return res.json(err)
+  }
+})
+
+app.get('/:installationId/repoSyncState.json', async (req, res) => {
+  const { Subscription } = require('../models')
+  try {
+    const subscriptions = await Subscription.getAllForInstallation(req.params.installationId)
+    if (!subscriptions.length) {
+      return res.sendStatus(404)
+    }
+    const data = subscriptions[0].dataValues.repoSyncState
+    return res.json(data)
+  } catch (err) {
+    res.status(500)
+    return res.json(err)
+  }
+})
+
+app.post('/:installationId/sync', bodyParser, async (req, res) => {
+  const { Installation, Subscription } = require('../models')
+  const { installationId } = req.params
+  const { jiraHost } = req.body
+
+  const installation = await Installation.getForHost(jiraHost)
+
+  if (!installation) {
+    res.sendStatus(404)
+  } else {
+    try {
+      const subscription = await Subscription.getSingleInstallation(jiraHost, installationId)
```

Comment:
Could we get the installationId from the installation object and not from the query params?

Reply:
Hah, I had to double check this because of the poor naming choices that have plagued us.

So the installation object doesn't have the github installation ID. In this case `Installation` is the installation of the integration in Jira, so all we have available is the `jiraHost` and `clientKey`. One `jiraHost` can have many `Subscriptions` to various GitHub Installation IDs (different orgs or users). So the parameter here is a way to say "I want the status of the sync from GitHub Installation 12345 on Jira Host tcbyrd.atlassian.net", because you could also have the same GitHub Installation ID on multiple instances.

So TL;DR, we need a way to provide the github installation ID as a value, and the path param makes it feel "RESTful" to me. Thoughts?

Reply:
Ok. Makes sense.

But then the `const installation = await Installation.getForHost(jiraHost)` is not needed in this endpoint, right? Initially I thought it was to get the installationId, but I see that it's a different type of installation (jira vs github installation).

Reply:
Ah, yeh. Good point. I pulled this code over from another route that checks for the Installation and sends a `404` if there isn't one, but it would make sense here to send that if no Subscription is found for the installation ID. I'll update it!